process.go

```diff
@@ -58,12 +35,221 @@ type Process struct {
+	for _, volume := range copied.Volumes {
+		volumes = append(volumes, volume.Copy())
+	}
+	copied.Volumes = volumes
+
+	return copied
+}
+
+// ProcessFieldValue describes a requested change to a Process.
+type ProcessFieldValue struct {
+	Field    string
+	Subfield string
+	Value    string
+}
+
+// Override updates the Process with the provided value. If the
+// identified field is not already set then Override fails.
+func (p *Process) Override(value ProcessFieldValue) error {
+	switch value.Field {
+	case "name":
+		// TODO(ericsnow) Allow overriding the name (for multiple copies)?
+		return fmt.Errorf(`cannot override "name"`)
+	case "description":
+		if p.Description == "" {
+			return fmt.Errorf(`cannot override "description", not set`)
+		}
+		p.Description = value.Value
+	case "type":
+		return fmt.Errorf(`cannot override "type"`)
+	case "type-options":
+		if value.Subfield == "" {
+			return fmt.Errorf(`cannot override "type-options" without sub-field`)
+		}
+		if _, ok := p.TypeOptions[value.Subfield]; !ok {
+			return fmt.Errorf(`cannot override "type-options" field %q, not set`, value.Subfield)
+		}
+		p.TypeOptions[value.Subfield] = value.Value
+	case "command":
+		if p.Command == "" {
+			return fmt.Errorf(`cannot override "command", not set`)
+		}
+		p.Command = value.Value
+	case "image":
+		if p.Image == "" {
+			return fmt.Errorf(`cannot override "image", not set`)
+		}
+		p.Image = value.Value
+	case "ports":
+		if value.Subfield == "" {
+			return fmt.Errorf(`cannot override "ports" without sub-field`)
+		}
+		index, err := strconv.Atoi(value.Subfield)
+		if err != nil {
+			return fmt.Errorf(`"ports" sub-field must be an integer index`)
+		}
+		if index < 0 || index >= len(p.Ports) {
+			return fmt.Errorf(`"ports" index %d out of range`, index)
+		}
+		var port ProcessPort
+		if err := port.Set(value.Value); err != nil {
+			return err
+		}
+		p.Ports[index] = port
+	case "volumes":
+		if value.Subfield == "" {
+			return fmt.Errorf(`cannot override "volumes" without sub-field`)
+		}
+		index, err := strconv.Atoi(value.Subfield)
+		if err != nil {
+			return fmt.Errorf(`"ports" sub-field must be an integer index`)
+		}
+		if index < 0 || index >= len(p.Ports) {
+			return fmt.Errorf(`"ports" index %d out of range`, index)
+		}
+		var volume ProcessVolume
+		if err := volume.Set(value.Value); err != nil {
+			return err
+		}
+		p.Volumes[index] = volume
+	case "env":
+		if value.Subfield == "" {
+			return fmt.Errorf(`cannot override "env" without sub-field`)
+		}
+		if _, ok := p.EnvVars[value.Subfield]; !ok {
+			return fmt.Errorf(`cannot override "env" field %q, not set`, value.Subfield)
+		}
+		p.EnvVars[value.Subfield] = value.Value
+	default:
+		return fmt.Errorf("unrecognized field %q", value.Field)
+	}
+	return nil
+}
+
+// Extend updates the Process with the provided value. If the
+// identified field is already set then Extend fails.
+func (p *Process) Extend(value ProcessFieldValue) error {
```

Comment:
As a whole there are subtle differences which mean it's not quite the inverse.  I actually started off trying to do what you are talking about and it ended up messier and harder to follow.  This way is clear and explicit.  If it turns out to be a problem (which I doubt) then we can refactor these pieces later.